.github/workflows/README.md

```diff
@@ -133,4 +133,47 @@ Composite actions are a convenient way to reuse build logic, but they have
 some limitations. 
 
 - Cannot run more than one step in a composite action (see `workflow_call` instead)
-- Inputs can only be strings, no support for typed parameters. See: https://github.com/actions/runner/issues/2238
\ No newline at end of file
+- Inputs can only be strings, no support for typed parameters. See: https://github.com/actions/runner/issues/2238
+
+## Troubleshooting
+
+### Gradle Cache Misses
+
+If your PR is running for longer than you would expect due to cache misses, there are a
+few things to check. 
+
+First, find the cache that was loaded into your PR build. This is found in the Setup Gradle
+output. Look for a line starting with "Restored Gradle User Home from cache key". 
+For example,
+
+```
+Restored Gradle User Home from cache key: gradle-home-v1|Linux-X64|test[188616818c9a3165053ef8704c27b28e]-5c20aa187aa8f51af4270d7d1b0db4963b0cd10b
+```
+
+The last part of the cache key is the SHA of the commit on trunk where the cache
+was created. If that commit is not on your branch, it means your build loaded a 
+cache that includes changes your PR does not yet have. This is a common way to
+have cache misses. To resolve this, update your PR with the latest cached trunk commit:
+
+```commandline
+git fetch origin
+./committer-tools/update-cache.sh
```

Comment:
I noticed that this script requires us to login Github (or access key) - is it possible to use `curl + jq` to parse `https://api.github.com/repos/apache/kafka/actions/caches?key=gradle-home-v1&&ref=refs/heads/trunk`?

Reply:
Wow, I'm surprised that works! I would have assumed that even public APIs would need some kind of auth token.

I filed https://issues.apache.org/jira/browse/KAFKA-18903 for this